token/program-2022/src/extension/reallocate.rs

```diff
@@ -83,8 +84,23 @@ pub fn process_reallocate(
     )?;
 
     // unpack to set account_type, if needed
-    let mut token_account = token_account_info.data.borrow_mut();
-    set_account_type::<Account>(&mut token_account)?;
+    let mut token_account_data = token_account_info.data.borrow_mut();
+    set_account_type::<Account>(&mut token_account_data)?;
+
+    // sync the rent exempt reserve for native accounts
+    let mut token_account = StateWithExtensionsMut::<Account>::unpack(&mut token_account_data)?;
+    if token_account.base.is_native.is_some() {
```

Comment:
nit: seems `token_account.base.is_native()` is more idiomatic across codebase.

Reply:
+1